src/main/java/com/marathon/hazelcast/servicediscovery/MarathonDiscoveryStrategy.java

```diff
@@ -38,16 +38,23 @@
 
     private int port;
 
+    private String marathonUsername;
+
+    private String marathonPassword;
+
     private ILogger logger;
 
     public MarathonDiscoveryStrategy(final ILogger logger, Map<String, Comparable> properties) {
         super(logger, properties);
         this.marathonEndpoint = getOrNull("discovery.marathon", MarathonDiscoveryConfiguration.MARATHON_ENDPOINT);
         this.appId = getOrNull("discovery.marathon", MarathonDiscoveryConfiguration.APP_ID);
         this.port = Integer.parseInt(getOrNull("discovery.marathon", MarathonDiscoveryConfiguration.PORT_INDEX));
+        this.marathonUsername = getOrNull("discovery.marathon", MarathonDiscoveryConfiguration.MARATHON_USERNAME);
+        this.marathonPassword = getOrNull("discovery.marathon", MarathonDiscoveryConfiguration.MARATHON_PASSWORD);
         this.logger = logger;
         try {
-            MarathonServiceDiscoveryHelper.start(marathonEndpoint, appId, port, logger);
+            MarathonServiceDiscoveryHelper.start(marathonEndpoint, appId, port, marathonUsername, marathonPassword,
+                    logger);
         } catch (Exception e) {
            logger.severe("Failed to start service discovery!", e);
```

Comment:
Is it ok to just log the exception? What is the client expectation?

Reply:
Service startup should not be impacted if marathon discovery fails so I guess that's why we are not rethrowing the Exception here.